bazel-ios/rules_ios
# Support dynamic framework/xcframework in Virtualized file system

Goal is to have VFS info correctly surface for static/dynamic frameworks, static library and xcframework in this PR.

### Changes made
1. Introduce a new attr for vfs_overlay rule to pass in content of swiftmodule. Right now the asusmption is that .swiftmodule is really a dir that contains .swiftinterface and .swiftdoc. This fixes normal static/dynamic framework not expressing swiftmodules in VFS 
2. In current _xcframework macro, added one more `select` which encapsulates the vfs overlay info that is based on the platform settings passed in. 

### How to test
Will add tests in future as we want to ensure VFS works for downstream project first as a standalone feature

TODO 
- [x] Rename swiftmodule -> swiftmodule_contents 
- [X] Handle the case when .swiftmodule is just a binary?  -> We now assume .swiftmodule binary is under .swiftmodule dir
